src/providers/UserProvider.tsx

```diff
@@ -40,8 +42,8 @@ export default function UserProvider({ children }: UserProviderProps): JSX.Eleme
 
       if (response.requestSucceeded) {
         setUser(response.user!);
-        if (response.user && !userState?.gtmInstrumented && (window as any).INIT_GTAG) {
-          setUserState({ gtmInstrumented: true });
+        if (response.user && !userAnalyticsState?.gtmInstrumented && (window as any).INIT_GTAG) {
+          dispatch(updateGtmInstrumented({ gtmInstrumented: true }));
```

Comment:
one last request, can we do an `await` on this? I'm guessing the previous recoil operation was not async. Here we don't want to initialize Google Analytics twice, if dispatch is async the code can theoretically get called twice (though it may not in the real world).